source/audio/appsVolume.py

```diff
@@ -0,0 +1,159 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2024 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+import config
+import globalVars
+from logHandler import log
+import nvwave
+from pycaw.utils import AudioSession
+import ui
+from dataclasses import dataclass
+from threading import Lock
+from config.featureFlagEnums import AppsVolumeAdjusterFlag
+from typing import NamedTuple
+from .utils import AudioSessionCallback, DummyAudioSessionCallback
+
+
+class VolumeAndMute(NamedTuple):
+	volume: float
+	mute: bool
+
+
+_appVolumesCache: dict[int, VolumeAndMute] = {}
+_appVolumesCacheLock = Lock()
+_activeCallback: DummyAudioSessionCallback | None = None
+
+
+def initialize() -> None:
+	state = config.conf["audio"]["applicationsVolumeMode"]
+	if state == AppsVolumeAdjusterFlag.MUTED:
+		state = AppsVolumeAdjusterFlag.ENABLED
+		config.conf["audio"]["applicationsVolumeMode"] = state
+	volume = config.conf["audio"]["applicationsSoundVolume"]
+	_updateAppsVolumeImpl(volume / 100.0, state)
+

```

Comment:
Here I disagree with the rabbit. This is needed to implement logic where muted state must not be persistent across restarts, but other state are persistent. IIRC @CyrilleB79 pushed for this behavior.

Reply:
Actually, I was pushing for neither state to be recorded nor restored: neither mute, nor volume.
But the discussion was so lengthy that I have not been able to follow up.

Maybe the best thing to do now is to release a first version of this feature in alpha, so that more people provide feedback.

Reply:
Ok, then to state my own opinion, there must be a way to make enabled state persistent -  as it would be too inconvenient to enable it manually each time NVDA restarts.